src/credentials/CHIPOpCred.h

```diff
@@ -0,0 +1,184 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines data types and objects for modeling and
+ *      working with CHIP Operational Credentials.
+ *
+ */
+
+#pragma once
+
+#include <credentials/CHIPCert.h>
+#include <crypto/CHIPCryptoPAL.h>
+#include <support/DLLUtil.h>
+
+namespace chip {
+namespace Credentials {
+
+using namespace Crypto;
+
+/**
+ *  @class ChipOperationalCredentialSet
+ *
+ *  @brief
+ *    Collection of CHIPCertificateSet providing methods for
+ *    finding and validating Device Credentials against Roof of Trust chains.
+ */
+class DLL_EXPORT ChipOperationalCredentialSet
+{
+public:
+    ChipOperationalCredentialSet();
+    ~ChipOperationalCredentialSet();
+
+    /**
+     * @brief Initialize ChipOperationalCredentialSet.
+     *        This initialization method is used when all memory structures needed for operation are
+     *        allocated externally and methods in this class don't need to deal with memory allocations.
+     *
+     * @param certSetsArray         A pointer to the array of the ChipCertificateSet structures.
+     * @param certSetsArraySize     Number of ChipCertificateSet entries in the array.
+     * @param deviceOpCredKeypair   A pointer to the Device Operational Credential's corresponding Keypair.
+     * @param deviceCredentialsDER  A pointer to the raw data buffer in DER format for the Device Operational Credentials.
+     * @param deviceCredentialsLen  Length in bytes of the Device Operational Credentials.
+     *
+     * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR Init(ChipCertificateSet * certSetsArray, uint8_t certSetsArraySize, P256Keypair * deviceOpCredKeypair,
+                    const uint8_t * deviceCredentialsDER, uint16_t deviceCredentialsLen);
+
+    /**
+     * @brief Find certificate set in the set.
+     *
+     * @param trustedRootId  Authority key identifier of the root certificate-set to be found in the operational credential set.
+     *
+     * @return A pointer to the certificate data On success. Otherwise, NULL if no certificate found.
+     **/
+    ChipCertificateSet * FindCertSet(const CertificateKeyId & trustedRootId) const;
+
+    /**
+     * @return A pointer to the set of certificate-set data entries.
+     **/
+    const ChipCertificateSet * GetOpCredSet() const { return mOpCreds; }
+
+    /**
+     * @return A pointer to the last certificate-set in the set. Returns NULL if certificate set is empty.
```

Comment:
What is the use for getting the last entry?

The restriction of 'may be null' seem to require extra logic in any users of the class.
I would suggest to remove this and then add it (likely modified without the null requirement) as a method when it is required.

Reply:
Will do so.